### DISCUSSION TITLE
Potential bug in stt/flt?

### DISCUSSION THREAD

**Discussion by jkim971201** [The-OpenROAD-Project/OpenROAD-flow-scripts#1697] [Ideas]
If FLUTE_REMOVE_DUPLICATE_PIN is turned on, then the flute code detects if there are duplicate pins [here](https://github.com/The-OpenROAD-Project/OpenROAD/blob/ead3236e56d57362a8dc8b25658fb41a191eba17/src/stt/src/flt/flute.cpp#L1088C1-L1100C7).

However, I think this can only detect a sinlge pair of duplicate pins. 
If there are multiple pairs of duplicate pins, then it will fail to remove all the duplicate pairs and produce some bugs in the flute_RDP(). 
I do not know we have to consider multiple duplicate pin pairs in the OpenROAD flow, 
but it will make an error if we use stt/flt as an independent library.  
(besides, I think the xs, ys, s vectors should be resized after duplicate removal because [flutes()](https://github.com/The-OpenROAD-Project/OpenROAD/blob/ead3236e56d57362a8dc8b25658fb41a191eba17/src/stt/include/stt/flute.h#L151C3-L151C3) uses the size of the vector as a new net degree.)

Please let me know if I'm misunderstanding.
Thanks.

**Comment by maliberty** [comment]
Currently we don't use that option (https://github.com/The-OpenROAD-Project/OpenROAD/blob/ead3236e56d57362a8dc8b25658fb41a191eba17/src/stt/include/stt/flute.h#L52) so I don't think it can have any effect on the OR flow.  I expect the bug comes from the original flute code that we imported.  

If it is important to you then you are welcome to try to fix it and make a PR.  However I don't think that will have any direct effect on OR due to our not using this option.
